components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/endpoint/registry/impl/EndpointRegistryImpl.java

```diff
@@ -190,6 +196,11 @@ public EndpointRegistryEntry getEndpointRegistryEntryByUUID(String registryId, S
      */
     public String addEndpointRegistryEntry(EndpointRegistryEntry registryEntry) throws EndpointRegistryException {
 
+        // If the displayName is empty, set the entryName as the displayName
+        if (StringUtils.isEmpty(registryEntry.getDisplayName())) {
```

Comment:
since we have done the same logic in multiple places, shall we move this logic to some common place, like mappingUtils?

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/pull/8645/commits/56072355e9225b17e47a17ba49e7d6fbf0c6539c